### DISCUSSION TITLE
How to customize/change toolboxes? Add links to e.g. brightness and contrast, or sharpen?

### DISCUSSION THREAD

**Discussion by pstein** [tannerhelland/PhotoDemon#511] [General]
Is there a way to customize the toolbox sub panes. on the left size?

I want e.g. to add a shortcut symbols to
- `brightness and contrast`
and
- `sharpen `
in one of these toolboxes.

How does that work?

Putting these shortcut symbols into horizontal toolbar would be acceptable as well.

**Comment by hi5** [comment]
As far as I know that is not yet possible. (Being able to change / assign Keyboard shortcuts is on the roadmap it seems https://github.com/tannerhelland/PhotoDemon/issues/474 but may take some time).

I would suggest to allow the user to create their own toolbar (section) where they could assign all available commands/user actions/macros to a "button":

(+ mandatory, - optional)

`+` Action
`-` Keyboard Shortcut
`-` Tooltip (if omitted name: of Action or Macro file)
`-` Icon (by default a generic "tool" icon with numbers if none is chosen, see below, otherwise file picker for ICO/PNG)


Example default icon with numbers:
![user-toolbar](https://github.com/tannerhelland/PhotoDemon/assets/1973899/a970047a-47b8-453a-89c0-afb58c84441c)
